src/sinks/amqp/config.rs

```diff
@@ -29,6 +34,9 @@ impl AmqpPropertiesConfig {
         if let Some(content_encoding) = &self.content_encoding {
             prop = prop.with_content_encoding(ShortString::from(content_encoding.clone()));
         }
+        if self.persistent {
```

Comment:
It's not obvious why we set the delivery mode to 2 to enable persistence.

Can you add a comment here with a link to https://www.rabbitmq.com/publishers.html#message-properties (or somewhere better if you can find better docs about the delivery mode) to explain what the 2 means.